File: LeetCode/Easy/0066-plus-one/0066-plus-one.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
    let n = digits.length - 1;

    while (digits[n] == 9 & n > 0){
        digits[n] = 0;
        n -= 1;
    }

    if (n == 0 & digits[n] == 9){
        digits[n] = 1;
        digits.push(0);
        return digits;
    }

    digits[n] ++;
    return digits;
    
};

File: LeetCode/Easy/0066-plus-one/0066-plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        y = (len(digits) - 1)

        while y != 0 and digits[y] == 9:
            digits[y] = 0
            y -= 1

        if digits[y] == 9:
            digits[y] = 1
            digits.append(0)
            return digits
       
        digits[y] += 1
        return digits